trpcultivate_germplasm/src/Plugin/TripalImporter/GermplasmCollectionImporter.php

```diff
@@ -0,0 +1,1169 @@
+      }
+      else {
+        $messages['valid_delimited_file']['status'] = 'pass';
+      }
+    }
+
+    // Call the processListWithDescribedTable() method in ValidHeaders class
+    // to check if there are any failures for the valid headers validator.
+    if (array_key_exists('valid_headers', $failures)) {
+      if (!empty($failures['valid_headers'])) {
+        $messages['valid_headers']['status'] = 'fail';
+
+        // Configure the metadata.
+        $metadata = [
+          'column_headers' => $header_names,
+        ];
+        $messages['valid_headers']['details'] = ValidHeaders::processListWithDescribedTable($failures['valid_headers'], $metadata);
+      }
+      else {
+        $messages['valid_headers']['status'] = 'pass';
+      }
+    }
+
+    // Call the processListWithDescribedTable() method in EmptyCell class to
+    // check if there are any failures for the empty cell validator.
+    if (array_key_exists('empty_cell', $failures)) {
+      if (!empty($failures['empty_cell'])) {
+        $messages['empty_cell']['status'] = 'fail';
+        // Configure the metadata.
+        $metadata = [
+          'column_headers' => [
+            0 => $header_names[0],
+            1 => $header_names[1],
+            2 => $header_names[2],
+          ],
+        ];
+        $messages['empty_cell']['details'] = EmptyCell::processListWithDescribedTable($failures['empty_cell'], $metadata);
+      }
+      elseif (!$raw_row_failed) {
+        $messages['empty_cell']['status'] = 'pass';
+      }
+    }
+
+    // Call the processListWithDescribedTable() method in GermplasmNameExists
+    // class to check if there are any failures for the germplasm name exists
+    // validator.
+    if (array_key_exists('germplasm_name_exists', $failures)) {
+      if (!empty($failures['germplasm_name_exists'])) {
+        $messages['germplasm_name_exists']['status'] = 'fail';
+        // Configure the metadata.
+        $metadata = [
+          'column_headers' => [
+            0 => $header_names[0],
+          ],
+        ];
+        $messages['germplasm_name_exists']['details'] = GermplasmNameExists::processListWithDescribedTable($failures['germplasm_name_exists'], $metadata);
+      }
+      elseif (!$raw_row_failed) {
+        $messages['germplasm_name_exists']['status'] = 'pass';
+      }
+    }
+
+    return $messages;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function formValidate($form, &$form_state) {
+
+    $form_values = $form_state->getValues();
+
+    // Validate Population Entry.
+    $fld_name_population_entry = 'fld_text_population_entry';
+    $fld_value_population_entry = $form_values[$fld_name_population_entry];
+
+    $parsed_stock_id = ChadoGenericAutocompleteController::getPkeyId($fld_value_population_entry);
+
+    // Failed to locate the poplation stock field element.
+    if ($parsed_stock_id == 0) {
+      throw new \Exception('Germplasm does not exist. Please enter a valid germplasm in Population Entry.');
```

Comment:
```suggestion
     $form_state->setErrorByName($fld_name_population_entry, 'Germplasm does not exist. Please enter a valid germplasm in Population Entry.');
```
A handy method from Drupal to highlight a field with unexpected value.